### ISSUE TITLE
False positive: unnecessary semicolon

### ISSUE THREAD

**Issue by JakeWharton** [pinterest/ktlint#703]
```kotlin
val keyFactory = run {
  val random = Random(1);
  { random.nextInt() }
}
```
This is a minimal reproducing sample which is contrived, not the actual code.

The problem is that without the semicolon Kotlin thinks that I'm calling the `Random` constructor with a trailing lambda and fails to compile. Thus, the semicolon is necessary to separate the statement from the lambda (which is an implicit return value).

**Comment by rom4ek** [comment]
Hm, I just tried to run `no-semi` rule over the sample, but it works just fine.
```
    @Test
    fun `test`() {
        assertThat(
            NoSemicolonsRule().lint(
                """
                val keyFactory = run {
                  val random = Random(1);
                  { random.nextInt() }
                }
                """.trimIndent()
            )
        ).isEmpty()
    }
```

Also, looks like it's explicitly handled on line 59:
https://github.com/pinterest/ktlint/blob/cb0595b30d23fea7f005686d46c2c14fd8206d40/ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/NoSemicolonsRule.kt#L54-L63

Am I missing something, or the actual code was a bit different?

**Comment by shashachu** [comment]
Closing until we get more information.

**Comment by Mygod** [comment]
More false positive edge cases like this: https://github.com/Mygod/VPNHotspot/blob/e333c859dfe4e675005a56049c510e5fb4ff270c/mobile/src/main/java/be/mygod/librootkotlinx/RootServer.kt#L498

**Comment by romtsn** [comment]
@Mygod so the semicolon is necessary there, right? Mind filing a new issue?
